numpy/core/src/multiarray/numpyos.c

```diff
@@ -580,7 +580,12 @@ NumPyOS_ascii_strtold(const char *s, char** endptr)
     }
     return result;
 #else
-    return NumPyOS_ascii_strtod(s, endptr);
+    errno = 0;
+    result = strtold(s, endptr);
```

Comment:
this needs a `HAVE_` check too as not all platforms will have strotold, add it to OPTIONAL_STDFUNCS in  `numpy/core/setup_common.py` to generate the define